CMakeLists.txt

```diff
@@ -28,5 +31,8 @@ file(GLOB SOURCE_FILES "src/*.cpp" "src/*.h")
 # Declare target executable
 add_executable(${TARGET} ${SOURCE_FILES})
 
+# Link libraries
+target_link_libraries(${TARGET} template-library)
```

Comment:
@rpannek well spotted, the new patch set fixes this issues as per our offline discussion.